active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,230 @@
+
+To address this issue, the deployments and deployment operations APIs must be updated to include additional information, such as:
+
+- Resource symbolic name
+- Resource type
+- Deployment provider alias
+- Deployment provider name
+- Deployment provider version
+
+This additional data would enable users to identify a deployed extensible resource and map it to the corresponding resource definition in a Bicep file or an ARM template, providing users with comprehensive insights.
+
+## Detailed design
+
+> The core of the REP. Elaborate on the design and implementation with sufficient detail for someone familiar with Bicep to grasp. Provide detailed examples to illustrate how the feature is used and its implications on user experience.
+
+### `Microsoft.Resources/deployments` API changes for PUT and GET
+
+#### Adding `deploymentProviders`
+
+The `Microsoft.Resources/deployments` API will be updated to include a `deploymentProviders` array in both the PUT and GET response bodies. This modification is intended to incorporate information about deployment providers that are involved in executing the deployment to offer a more comprehensive deployment view.
+
+> It's worth noting that we cannot use the name `providers` as the property `providers` is already assigned to return Azure Resource Provider information. While an alternative approach could involve renaming the current `providers` property to `resourceProviders` for enhanced accuracy, the solution should be avoided as it constitutes a breaking change.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    ...
+    "providers": [
+      {
+        "namespace": "Microsoft.Storage",
+        "resourceTypes": [
+          {
+            "resourceType": "storageAccounts",
+            "locations": [
+              "eastus"
+            ]
+          }
+        ]
+      }
+    ],
++   "deploymentProviders": [
++     {
++       "alias": "az",
++       "name": "AzureResourceManager",
++       "version": "1.0.0"
++     }  ,
++     {
++       "alias": "kubernetes",
++       "name": "Kubernetes",
++       "version": "1.27.8"
++     }
+    ]
+  }
+}
+```
+
+#### Updating the `outputResources` to include more identifiable information
+
+To improve the current structure where each `outputResources` object only has a resource `id`, it's essential to add more properties for identifying extensible resources, which lack resource IDs. These additions should include the symbolic name, the resource type, and the deployment provider alias.
```

Comment:
> identifying extensible resources, which lack resource IDs

Doesn't the v2 contract change this?

Reply:
Yeah, I'm going to mention it in a follow-up REP to be sent out shortly. One question I have is whether we should reuse the property name `id`, as some clients may expect it always to be a valid resource ID, but for extensible resources ID is a freeform string.

Reply:
For context, orginally I wrote a single REP that covered both this feature and the stacks integration. However, as the document expanded, it became increasingly complex and challenging to navigate, given it essentially addressed two topics, so I decided to separate it into two for clarity.

Reply:
Wanted to revisit this in the discussion today. It feels like an "id" would be more useful for the purposes we use the `outputResources` field today (tracking changes to specific resources). Having the symbolic name would be useful for user understanding, but doesn't feel like it solves the same problem.

I think @rynowak's comment is related.